lib/publiccloud/aws_client.pm

```diff
@@ -80,9 +53,6 @@ sub init {
             $self->key_id($data->{access_key_id});
             $self->key_secret($data->{secret_access_key});
         }
-    } elsif (!defined($self->key_id) || !defined($self->key_secret)) {
```

Comment:
IMO you need to remove `if` part above also and  you need to change `get_credentials` from get_var to get_required_var

Reply:
Done 👍